### PR TITLE
Adapts package to new release process

### PR DESCRIPTION
**Pull request by ncordon**
* Adapts package to new release method.
* Adds exhaustive tests to check that files are being read correctly and generated tree has appropriate number of nodes.
* Adds docs for the package.

### CODE REVIEW

File: src/main/scala/helpers.scala [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  * - Method to check whether all the files for a [[Version]] exist in S3.
+  * - Wrappers for files utilities (uncompress, extract, download)
+  */
+private[ncbitaxonomy] case object helpers {
+
+  lazy val s3Client = AmazonS3ClientBuilder.standard().build()
+
+  val partSize5MiB = 5 * 1024 * 1024
+
+  /** Downloads the specified `s3Obj` to a give `file` */
```
Comment (agarciamontoro) [inline src/main/scala/helpers.scala]:
```suggestion
  /** Downloads the specified `s3Obj` to a given `file` */
```

File: src/main/scala/helpers.scala [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    request
+      .paranoidPutFile(s3Client)(file, s3Obj, partSize5MiB)(
+        data.hashingFunction
+      )
+      .left
+      .map { err =>
+        Error.S3Error(err)
+      }
+
+  /** Returns true when object does not exists or communication with S3
```
Comment (agarciamontoro) [inline src/main/scala/helpers.scala]: Typo?

```suggestion
  /** Returns true when object does exist or communication with S3
```

File: src/main/scala/io.scala [hunk 3]
```diff
@@ -186,19 +276,45 @@ case object io {
...
+    * val shapeFile = new File("./data/0.1.0/shape.tree")
+    *
+    * val treeResult = readTaxTreeFromFiles(dataFile, shapeFile)
+    *
+    * treeResult match {
+    *   case Left(error: FileError) => println(error.msg)
+    *   case Right(Left(error: SerializationError)) => println(error.msg)
+    *   case Right(Right(tree)) => // do stuff with the tree
+    * }
+    * }}}
```
Comment (agarciamontoro) [inline src/main/scala/io.scala:313]: I think this documentation is outdated, as the function returns Left(error) or Right(tree), isn't it?

File: src/main/scala/release.scala [hunk 4]
```diff
@@ -0,0 +1,111 @@
+package ohnosequences.db.ncbitaxonomy
+
+import ohnosequences.s3.S3Object
+import java.io.File
+import helpers._
+import data._
+
+case object release {
+
+  /** Downloads, extract and uncompress the file from [[remote.sourceFile]],
```
Comment (agarciamontoro) [inline src/main/scala/release.scala]: Typos

```suggestion
  /** Downloads, extracts and uncompresses the file from [[remote.sourceFile]],
```

File: src/main/scala/release.scala [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  private def getTaxDump(version: Version): Error + (File, File) = {
+    val taxDump   = local.taxDump(version)
+    val tarFile   = local.tarFile(version)
+    val nodesFile = local.nodes(version)
+    val namesFile = local.names(version)
+
+    for {
+      toUncompress <- downloadFromURL(remote.sourceFile, taxDump)
+      tarFile      <- uncompress(toUncompress, tarFile)
+      _            <- extract(tarFile, localFolder(version))
```
Comment (agarciamontoro) [inline src/main/scala/release.scala:30]: These two lines appear way too much in all the repos, so I've opened this: https://github.com/ohnosequences/files/issues/26

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
LGTM! I've left some comments below but it is really good, and the work on the documentation (we needed it) is great, thanks!

But: I've tested the code and the tests fail because there is an object whose metadata does not contain the key for the hash function. I don't know which object it is because the error in S3 is not that helpful (my bad, I'll fix it in https://github.com/ohnosequences/s3/issues/36)

**Comment by ncordon** [comment]
I think they are the old versions, sorry. I should have kept only 0.2.0 Version there, which is the only version with metadata.

File: src/main/scala/release.scala (hunk 4 above)
Comment (ncordon) [reply src/main/scala/release.scala]: Oh god, the third person :see_no_evil:

File: src/main/scala/io.scala (hunk 3 above)
Comment (ncordon) [reply src/main/scala/io.scala:313]: Yessss. Thanks

**Comment by ncordon** [comment]
Now everything works. When you give me the OK, I'll make a release of this and `db.taxonomy`

**Review by agarciamontoro — APPROVED** [review]
:tada: :tada: :tada: :tada:

**Pull request merged**